src/main/java/org/spongepowered/mod/network/SpongeModNetworkManager.java

```diff
@@ -68,54 +66,35 @@
 @Singleton
 public class SpongeModNetworkManager extends SpongeNetworkManager {
 
-    public static final EventContextKey<INetHandler> NET_HANDLER = new EventContextKey<INetHandler>() {
-        @Override
-        public Class<INetHandler> getAllowedType() {
-            return INetHandler.class;
-        }
+    private final Map<String, SpongeModChannelBinding> channelMap = Maps.newHashMap();
 
-        @Override
-        public String getId() {
-            return "sponge:nethandler";
-        }
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public void onCustomPacketRegistration(CustomPacketRegistrationEvent<?> event) {
+        final Set<String> channels = ((IMixinNetPlayHandler) event.getHandler()).getRegisteredChannels();
+        final EntityPlayer player;
 
-        @Override
-        public String getName() {
-            return "NetHandler";
+        if (event.getSide().isClient()) {
+            player = null; // TODO In 1.13, Client player will be a Sponge player, should we still put in the cause in 1.12?
+        } else {
+            player = ((NetHandlerPlayServer) event.getHandler()).player;
         }
-    };
 
-    private final Map<String, SpongeModChannelBinding> channelMap = Maps.newHashMap();
+        final Cause currentCause = Cause.of(EventContext.empty(), player == null ? Sponge.getGame() : player, event.getSide().isClient() ?
```

Comment:
Is is not on the main thread? Can't imagine why we wouldn't be able to use the cause stack manager to grab the parenting mod container.